### PR TITLE
perf: migrate link store to Rust

### PR DESCRIPTION
**Pull request by Wazzymandias**
## Motivation

- Move link store to rust for improved Hub performance

## Change Summary

- Add link store rust code
- Add convenience methods for Stores type conversions

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [x] PR has a [changeset](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#35-adding-changesets)
- [x] PR has been tagged with a change label(s) (i.e. documentation, feature, bugfix, or chore)
- [ ] PR includes [documentation](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#32-writing-docs) if necessary.
- [x] All [commits have been signed](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#22-signing-commits)

## Additional Context

If this is a relatively large or complex change, provide more details here that will help reviewers

---

## PR-Codex overview
This PR migrates the link store to Rust, adds new methods, implements traits, and refactors error handling.

### Detailed summary
- Migrated link store to Rust
- Added new methods to the link store
- Implemented traits for `IntoU8` and `IntoI32`
- Refactored error handling in the store
- Updated dependencies and imports

> The following files were skipped due to too many changes: `apps/hubble/src/rustfunctions.ts`, `apps/hubble/src/storage/stores/linkStore.ts`, `apps/hubble/src/addon/src/store/link_store.rs`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/hubble/src/addon/src/store/reaction_store.rs [hunk 1]
```diff
@@ -445,6 +440,25 @@ impl ReactionStore {
         ))))
     }
 
+    pub fn create_link_store(mut cx: FunctionContext) -> JsResult<JsBox<Arc<Store>>> {
```
Comment (sanjayprabhu) [inline apps/hubble/src/addon/src/store/reaction_store.rs:443]: This is in the wrong place?

File: apps/hubble/src/addon/src/store/link_store.rs [hunk 2]
```diff
@@ -0,0 +1,675 @@
...
+        r#type: String,
+        target: Option<Target>,
+    ) -> Result<Option<Message>, HubError> {
+        let partial_message = protos::Message {
+            data: Some(MessageData {
+                fid: fid as u64,
+                r#type: MessageType::LinkAdd.into(),
+                body: Some(message_data::Body::LinkBody(LinkBody{
+                    r#type,
+                    display_timestamp: None,
```
Comment (sanjayprabhu) [inline apps/hubble/src/addon/src/store/link_store.rs:105]: Why do we need `display_timestamp` here?

File: apps/hubble/src/addon/src/store/link_store.rs [hunk 3]
```diff
@@ -0,0 +1,675 @@
...
+    /// * `target` - target ID of the object being reacted to (currently just cast id)
+    /// * `fid` - the fid of the user who created the link
+    /// * `ts_hash` - the timestamp hash of the link message
+    fn links_by_target_key(
+        target: &Target,
+        fid: u32,
+        ts_hash: Option<&[u8; TS_HASH_LENGTH]>,
+    ) -> Vec<u8> {
+        if fid != 0 && (ts_hash.is_none() || ts_hash.is_some_and(|tsh| tsh.len() == 0)) {
+            panic!("bad_request.validation_failure - fid provided without timestamp hash");
```
Comment (sanjayprabhu) [inline apps/hubble/src/addon/src/store/link_store.rs:321]: I don't think we should panic in these functions?  We should return an Err with the right error code and message.

File: apps/hubble/src/addon/src/store/link_store.rs (hunk 2 above)
Comment (Wazzymandias) [reply apps/hubble/src/addon/src/store/link_store.rs:105]: the compiler throws error without full type definition, but I can do this differently

File: apps/hubble/src/addon/src/store/link_store.rs (hunk 3 above)
Comment (Wazzymandias) [reply apps/hubble/src/addon/src/store/link_store.rs]: fair enough, updated

File: apps/hubble/src/addon/src/store/link_store.rs (hunk 2 above)
Comment (Wazzymandias) [reply apps/hubble/src/addon/src/store/link_store.rs]: updated

File: apps/hubble/src/addon/src/store/reaction_store.rs (hunk 1 above)
Comment (Wazzymandias) [reply apps/hubble/src/addon/src/store/reaction_store.rs]: good catch, brain fart, removed

**Review by sanjayprabhu — APPROVED** [review]

**Pull request merged**